pkg/controller/daemon/daemon_controller.go

```diff
@@ -1421,6 +1374,15 @@ func NewPod(ds *apps.DaemonSet, nodeName string) *v1.Pod {
 	newPod := &v1.Pod{Spec: ds.Spec.Template.Spec, ObjectMeta: ds.Spec.Template.ObjectMeta}
 	newPod.Namespace = ds.Namespace
 	newPod.Spec.NodeName = nodeName
+
+	// NOTES (k82cn): Cherrypick (#68539), 'kubelettypes.IsCriticalPod' considered
+	// `features.ExperimentalCriticalPodAnnotation` in 1.12, but not in 1.11.
+	isCritical := kubelettypes.IsCriticalPod(newPod) &&
+		utilfeature.DefaultFeatureGate.Enabled(features.ExperimentalCriticalPodAnnotation)
```

Comment:
The criteria for `isCritical` is different here. 
In 1.11, as long as `ExperimentalCriticalPodAnnotation` is disabled, the pod can't be critical.
However, in 1.12, the pod might still be critical if `PodPriority` is enabled, even if  `ExperimentalCriticalPodAnnotation` is disabled.

1.12: https://github.com/kubernetes/kubernetes/blob/5be0a0ee8d16fc276124d69b0fa9d96b6dbd3f2a/pkg/kubelet/types/pod_update.go#L148-L160
1.11:
https://github.com/kubernetes/kubernetes/blob/38d92afc24b8ce6c7d1ca3cf04d395b21ae9ad39/pkg/kubelet/types/pod_update.go#L146-L148

Reply:
yes, so I check it with feature gate, and give a note here. this is also outofdisk ut for it.

Reply:
My question is - 
1.11: `ExperimentalCriticalPodAnnotation` feature && (critical annotations || pod priority)
1.12: (`PodPriority` feature && pod priority) || (`ExperimentalCriticalPodAnnotation` feature && critical annotations) 

They're quite different. 

Is it safe to ignore those differences? Will we miss some critical pods?

Reply:
Got your point, we need to handle that.  Do you think we should also cherry pick #66082, or we handle it here without cherry pick.